packages/drivers/routerlicious-driver/src/telemetry.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryGenericEvent, ITelemetryLogger } from "@fluidframework/common-definitions";
+
+/**
+ * Helper class to reduce telemetry noise by aggregating performance events.
+ * Aggregate data will be logged, on flush, as sums along with the total count.
+ * Optionally, takes a threshold param and will flush each time the threshold is reached.
+ */
+export class AggregatePerformanceEvent {
+    private data: Record<string, number> = {};
+    private count: number = 0;
+
+    constructor(
```

Comment:
i think this should just take a logger like performance event. it will make the interface a bit cleaner.

Reply:
PerformanceEvent is guaranteed to be instantiated where it is being used. However AggregatePerformanceEvent is not, and therefore is not guaranteed to have access to the correct logger, or any logger, at the time of instantiation.

Specifically in this context, the AggregatePerformanceEvents are being instantiated in the DocumentServiceFactory, which has no logger available, and any DocumentService created by the factory could have different loggers.

Reply:
i would change the architecture that the event and logger are owned at the same layer. my above comments will move you in that direction. the current architecture is unnecessarily complex.